File: 351003/Pisarchik/notebook/publisher/src/main/java/com/lizaveta/notebook/repository/CrudRepository.java
package com.lizaveta.notebook.repository;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.Optional;

public interface CrudRepository<T, ID> {

    Optional<T> findById(ID id);

    List<T> findAll();

    Page<T> findAll(Pageable pageable);

    T save(T entity);

    T update(T entity);

    boolean deleteById(ID id);

    boolean existsById(ID id);
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/repository/CrudRepository.kt
package com.example.distcomp.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface CrudRepository<T> {
    fun save(entity: T): T
    fun findById(id: Long): T?
    fun findAll(): List<T>
    fun findAll(pageable: Pageable): Page<T>
    fun deleteById(id: Long): Boolean
    fun existsById(id: Long): Boolean
}
